sentry_streams/sentry_streams/pipeline/pipeline.py

```diff
@@ -198,3 +209,49 @@ class Reduce(WithInput, Generic[MeasurementUnit, InputType, OutputType]):
     aggregate_backend: Optional[AggregationBackend[OutputType]] = None
     group_by_key: Optional[GroupBy] = None
```

Comment:
Why not adding a group_by_key abstract method in Reduce so that you do not have to do something like this in the Flink PR :
```
group_by = step.group_by_key if hasattr(step, "group_by_key") else None
```